client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -379,6 +362,18 @@ public void appendSetup(AppendSetup appendSetup) {
             }
         }
 
+        private void retryOnWriteFail(WireCommand wireCommand) {
+            if (state.encounteredSeal()) {
+                Retry.indefinitelyWithExpBackoff(retrySchedule.getInitialMillis(), retrySchedule.getMultiplier(),
+                        retrySchedule.getMaxDelay(),
+                        t -> log.error("Error while invoking SegmentSealed call back for writer: {}", writerId, t))
```

Comment:
Another instance of the naming being awkward, this code can be invoked by `noSuchSegment` as well.

Reply:
updated the code.